neo/Network/RPC/WalletAPI.cs

```diff
@@ -0,0 +1,190 @@
+        /// <param name="toAddress">address or account script hash</param>
+        /// <param name="amount">token amount</param>
+        /// <param name="networkFee">netwotk fee, set to be 0 will auto calculate the least fee</param>
+        /// <returns></returns>
+        public Transaction Transfer(string tokenHash, string fromKey, string toAddress, decimal amount, decimal networkFee = 0)
+        {
+            UInt160 scriptHash = tokenHash.ToUInt160();
+            var decimals = nep5API.Decimals(scriptHash);
+
+            KeyPair from = fromKey.ToKeyPair();
+            UInt160 to = toAddress.ToUInt160();
+            BigInteger amountInteger = amount.ToBigInteger(decimals);
+            BigInteger networkFeeInteger = networkFee.ToBigInteger(NativeContract.GAS.Decimals);
+            return Transfer(scriptHash, from, to, amountInteger, (long)networkFeeInteger);
+        }
+
+        /// <summary>
+        /// Transfer NEP5 token balance
+        /// </summary>
+        /// <param name="scriptHash">contract script hash</param>
+        /// <param name="fromKey">from KeyPair</param>
+        /// <param name="to">to account script hash</param>
+        /// <param name="amount">transfer amount</param>
+        /// <param name="networkFee">netwotk fee, set to be 0 will auto calculate the least fee</param>
+        /// <returns></returns>
+        public Transaction Transfer(UInt160 scriptHash, KeyPair from, UInt160 to, BigInteger amountInteger, BigInteger networkFeeInteger = default)
+        {
+            Transaction transaction = nep5API.CreateTransferTx(scriptHash, from, to, amountInteger, (long)networkFeeInteger);
+            rpcClient.SendRawTransaction(transaction);
+            return transaction;
+        }
+
+        /// <summary>
+        /// Wait until the transaction is observable block chain
+        /// </summary>
+        /// <param name="transaction">the transaction to observe</param>
+        /// <param name="timeout">TimeoutException throws after "timeout" seconds</param>
+        /// <returns>the Transaction state, including vmState and blockhash</returns>
+        public async Task<RpcTransaction> WaitTransaction(Transaction transaction, int timeout = 60)
+        {
+            DateTime deadline = DateTime.UtcNow.AddSeconds(timeout);
+            RpcTransaction rpcTx = null;
+            while (rpcTx == null || rpcTx.Confirmations == null)
+            {
+                if (deadline < DateTime.UtcNow)
+                {
+                    throw new TimeoutException();
+                }
+
+                try
+                {
+                    rpcTx = rpcClient.GetRawTransaction(transaction.Hash.ToString());
+                    if (rpcTx == null || rpcTx.Confirmations == null)
+                    {
+                        await Task.Delay(1000);
```

Comment:
BlockTime/2 ?

Reply:
👌This will reduce the `RPC` request times.